### ISSUE TITLE
[iRO] Packet Tokenizer: Unknown switch 09DB, 09DD

### ISSUE THREAD

**Issue by indroe** [OpenKore/openkore#351]
hi, today after maintenance i've got message like this,

Packet Tokenizer: Unknown switch: 09DB
Packet Tokenizer: Unknown switch: 09DD

and bot wont attacking mobs,

server iRO - Chaos

**Comment by xander30** [comment]
Packet Tokenizer: Unknown switch: 09DB
Packet Tokenizer: Unknown switch: 09DD

same here, any fix for this? bot wont attack just linger and walk

server iRO - Thor

**Comment by AranGala** [comment]
Yes, help please, same problem.

Packet Tokenizer: Unknown switch: 09DB
Packet Tokenizer: Unknown switch: 09DD

server iRO - Chaos

**Comment by rbaye1123** [comment]
I tried fixing the problem myself but none of the packet length extractors seem to be working.
http://openkore.com/index.php/Packet_Length_Extractor

Looks like we need a brand new recvpackets.txt... can somebody extract Rag.exe.EXE for us?

I attached newly updated Ragexe.EXE file.



[Ragexe.zip](https://github.com/OpenKore/openkore/files/641273/Ragexe.zip)

**Comment by indroe** [comment]
yeah, me too.. i try to fix it for myself using many tool, but i don't understand, any tool not working,,

**Comment by Sabog29** [comment]
any news?

**Comment by lututui** [comment]
I've got a new recvpackets for ya.

**Although I have not tested** (I don't have an iRO account and didn't bother to register), the packets 09DB and 09DD are listed, but this won't solve the whole problem: openkore doesn't know what they're used for.

Please note that the following changes probably **did not** happen all at once: the recvpackets for iRO is over 3 years old. Most of these packets do not change openkore's behavior at all, they are likely to be related to the client (visual effects, buttons etc). Except for some which do.

The recvpackets at the end of this comment should solve the Packet Tokenizers, but Packet Parsers will emerge, since openkore does not know what 09DB and 09DD are.

**Changes**
```
09BA 6 6 0 became 09BA 2 2 0
09BE 6 6 0 became 09BE 2 2 0
09CA 23 23 0 became 09CA -1 23 1
0A09 45 45 0 became 0A09 45 45 1
0A0A 47 47 0 became 0A0A 47 47 1
0A0B 47 47 0 became 0A0B 47 47 1
0A0C 56 56 0 became 0A0C 56 56 1
0A0D -1 4 0 became 0A0D -1 4 1
0A0F -1 4 0 became 0A0F -1 4 1
0A10 -1 4 0 became 0A10 -1 28 1
0A2D -1 4 0 became 0A2D -1 45 0
0A3B -1 7 1 became 0A3B -1 12 1
```

**New stuff**
```
09CC -1 4 0
09CD 8 8 1
09CE 102 102 1
09CF -1 4 0
09D0 -1 4 0
09D1 14 14 1
09D2 -1 28 0
09D3 -1 28 0
09D4 2 2 0
09D5 -1 4 1
09D6 -1 4 1
09D8 2 2 1
09D7 -1 5 1
09D9 4 4 0
09DA -1 8 0
09DB -1 84 1
09DC -1 77 1
09DD -1 78 1
09DE -1 33 1
09DF 7 7 1
09E0 -1 40 0
09E1 8 8 1
09E2 8 8 1
09E3 8 8 1
09E4 8 8 1
09E5 18 18 1
09E6 22 22 1
09E7 3 3 1
09E8 11 11 1
09E9 2 2 1
09EE 11 11 1
09EF 11 11 1
09F0 -1 7 1
0A7D -1 7 1
09F5 11 11 1
09F6 11 11 1
09EA 11 11 1
09EB -1 24 1
09F7 75 75 1
09EC -1 64 1
09ED 3 3 1
09F3 11 11 1
09F4 12 12 1
09F1 11 11 1
09F2 12 12 1
0A04 6 6 1
0A05 53 53 1
0A06 6 6 1
0A07 9 9 1
0A03 2 2 1
0A08 26 26 1
0A12 27 27 1
0A13 26 26 1
0A14 10 10 1
0A51 34 34 1
0A32 2 2 0
09F8 -1 8 1
09F9 143 143 1
09FA -1 6 1
09FB -1 6 0
09FC 6 6 1
09FD -1 86 1
09FE -1 79 1
09FF -1 80 1
0A00 269 269 1
0A01 3 3 0
0A02 4 4 1
0A11 -1 28 1
0A0E 14 14 1
0A15 12 12 1
0A16 26 26 0
0A17 6 6 1
0A18 14 14 1
0A19 2 2 1
0A1A 23 23 1
0A1B 2 2 1
0A1C -1 8 1
0A1D 2 2 1
0A1E 3 3 1
0A1F 2 2 1
0A20 21 21 1
0A21 3 3 1
0A22 5 5 1
0A23 -1 22 1
0A24 66 66 1
0A25 6 6 1
0A26 7 7 1
0A27 8 8 1
0A28 3 3 1
0A29 6 6 0
0A2A 6 6 0
0A2B 10 10 0
0A2C 12 12 0
0A2E 6 6 1
0A2F 7 7 1
0A30 106 106 1
0A31 -1 6 0
0A33 7 7 0
0A34 6 6 0
0A35 4 4 0
0A36 7 7 1
0A37 59 59 1
0A38 3 3 1
0A68 3 3 1
0A39 36 36 1
0A3A 12 12 1
0A3C -1 18 0
0A3D 18 18 0
0A3E -1 16 0
0A3F 9 9 0
0A40 11 11 0
0A41 18 18 0
0A42 43 43 0
0A43 85 85 0
0A44 -1 28 0
0A46 14 14 0
0A47 3 3 0
0A48 2 2 0
0A49 22 22 0
0A4A 6 6 0
0A4B 22 22 0
0A4C 28 28 0
0A8F 2 2 0
0A90 3 3 0
0A4D -1 64 0
0A79 -1 67 0
0A4E 4 4 0
0A70 2 2 0
0A4F -1 6 0
0A50 4 4 0
0A52 20 20 0
0A53 10 10 0
0A54 -1 8 0
0A55 2 2 0
0A56 6 6 0
0A57 6 6 0
0A58 8 8 0
0A59 -1 8 0
0A5A 2 2 0
0A5B 7 7 0
0A5C 18 18 0
0A5D 6 6 0
0A69 6 6 0
0A6A 12 12 0
0A6B -1 8 0
0A6C 7 7 0
0A6D -1 4 0
0A6E -1 68 1
0A6F -1 10 1
0A71 -1 5 0
0A72 61 61 0
0A73 2 2 0
0A74 8 8 0
0A76 80 80 0
0A77 15 15 0
0A78 15 15 0
0A7B -1 4 0
0A7C -1 4 0
0A7E -1 5 1
0A8C 2 2 1
0A80 6 6 1
0A7F -1 88 1
0A8D -1 17 1
0A81 4 4 1
0A92 -1 96 1
0A91 -1 25 1
0A93 3 3 1
0A94 2 2 1
0A89 57 57 1
0A8A 6 6 1
0A82 46 46 1
0A83 46 46 1
0A84 94 94 1
0A85 82 82 1
0A86 -1 4 1
0A87 -1 4 1
0A88 2 2 0
0A8B 2 2 0
0A8E 2 2 0
0A95 4 4 1
0A96 51 51 0
0A97 8 8 0
0A98 12 12 0
0A99 8 8 0
0A9A 10 10 0
0A9B -1 4 0
0A9C 2 2 0
0A9D 4 4 0
0A9E 2 2 0
0A9F 2 2 0
0AA0 2 2 0
0AA1 4 4 0
0AA2 -1 7 0
0AA3 7 7 0
0AA4 2 2 0
0AA5 -1 4 1
0AA6 36 36 1
0AA7 6 6 1
0AA8 5 5 1
0AA9 -1 4 1
0AAA -1 4 1
0AAB -1 5 1
0AAC 67 67 1
```

The whole file: [recvpackets.txt](https://github.com/OpenKore/openkore/files/641889/recvpackets.txt)

**Comment by rbaye1123** [comment]
Yeah I just tested them and it is exactly what she stated. It seems like the whole configuration has been changed... Openkore no longer recognizes any mosnters or items but are listed as numbers..  Looks like it needs a brand new makeup...

**Comment by xander30** [comment]
where do i paste this file sirs?

**Comment by Jyuchard** [comment]
Tested. Openkore still couldn't recognized the monster and just walking around.

**Comment by rbaye1123** [comment]
Openkore- Tables - iRO... but like i said it won't work. there will be another problem emerged after it being resolved. no monters and items are recognzieable and are only be listed as numbers, so your bot won't be doing much... Found out the hard way....

**Comment by xander30** [comment]
thanks i tried it and it doesnt work :(

**Comment by rbaye1123** [comment]
yeah.. Packet Parser : Unknown Switch:  09DB and 09DD

**Comment by xander30** [comment]
the bot is just walking around the map not attacking or looting :(

**Comment by rbaye1123** [comment]
@lututui yeah there has been changes in buttons and other various visual things from clients. do you think it is possible that iRO needs a whole another make up of openkore?
Just curious what is your take on it..

**Comment by bradley02** [comment]
this is somewhat related to this:
https://github.com/OpenKore/openkore/pull/275

**Comment by xander30** [comment]
iro updated the old mail system into Rodex mail system thats when the packets went insane :(

**Comment by gospelkiru** [comment]
hooo. maybe its a strategy so they can stop the bot'ers

**Comment by bradley02** [comment]
i think it is not the mail system but the charater code thing...

**Comment by indroe** [comment]
i just try to use recvpackets.txt from @lututui then the warning change become

Packet Parser: Unknown switch: 09DB
Packet Parser: Unknown switch: 09DD

so i think recvpacket problem solved, but openkore still doesn't understand, and kore still doesn't attack mobs,

**Comment by Henrybk** [comment]
Would you guys please test: https://github.com/Henrybk/openkore/tree/iRO-packet-bugfix

I used lututui recvpackets and my old iRO.pm I made some changes to, may work for you :P

**Comment by Henrybk** [comment]
Using the above said branch I was able to parse players names perfectly, I didn't try to do anything other than that so you may run into other bugs :)
Proof: https://i.gyazo.com/4bf96e538a577454466987c41b135ff0.png

**Comment by rbaye1123** [comment]
@Henrybk i am trying it out now let you know if it works!! Thanks

**Comment by rbaye1123** [comment]
@Henrybk Works problem solved. Thanks!

**Comment by indroe** [comment]
why i got error like this? This program has encountered an unexpected problem. This is probably because
of a recent server update, a bug in this program, or in one of the plugins.
We apologize for this problem. You may get support from IRC or the forums.

A detailed error report has been saved to errors.txt. Before posting a bug
report, please try out the SVN version first. If you are already using the SVN
version, search the forums first to see if your problem had already been solved,
or has already been reported. If you truly believe you have encountered a bug in
the program, please include the contents of the errors.txt in your bug report,
or we may not be able to help you!

The error message is:
Invalid arguments.

**Comment by Henrybk** [comment]
You guys should test #352, it probably is better than mine since I don't bot at all haha

**Comment by indroe** [comment]
#352 Worked!! so i close this issue, thanks guys

**Comment by bentot13** [comment]
why for me same error 
A detailed error report has been saved to errors.txt. Before posting a bug
report, please try out the SVN version first. If you are already using the SVN
version, search the forums first to see if your problem had already been solved,
or has already been reported. If you truly believe you have encountered a bug in
the program, please include the contents of the errors.txt in your bug report,
or we may not be able to help you!

**Comment by xander30** [comment]
hello how to fix this? i cant understand #352 post

**Comment by itsrachelfish** [comment]
@xander30 Just download the newest version from the main page: 

![download-zip](https://cloud.githubusercontent.com/assets/1670549/21054518/4dcd4acc-bdeb-11e6-821b-6b8de70b068c.png)

Going to lock this issue for now
